### DISCUSSION TITLE
debug mode ?

### DISCUSSION THREAD

**Discussion by simogeo** [publicodes/publicodes#399] [Ideas]
Bonjour,

existe t'il un mécanisme de _debug_ pour s'assurer que les calculs soient bien ceux attendus ? 

Merci de votre retour

**Comment by wiinxt** [comment]
Bonjour, si je ne dit pas de bêtise il n'existe pas de mécanisme `debug`, mais il existe un composant React de documentation automatique ici : https://publi.codes/docs/api/react-ui, c'est celui présent dans le bac à sable [sur la droite](https://publi.codes/studio/d%C3%A9penses-primeur#%0A%23%20Bienvenue%20dans%20le%20bac%20%C3%A0%20sable%20du%20langage%20publicode%20!%0A%23%20Pour%20en%20savoir%20plus%20sur%20le%20langage%20%3A%0A%23%20%3D%3E%20https%3A%2F%2Fpubli.codes%2Fdocs%2Fprincipes-de-base%0A%0Aprix%3A%0A%20%20avec%3A%0A%20%20%20%20carottes%3A%202%E2%82%AC%2Fkg%0A%20%20%20%20champignons%3A%205%E2%82%AC%2Fkg%0A%20%20%20%20avocat%3A%202%E2%82%AC%2Favocat%0A%0Ad%C3%A9penses%20primeur%3A%0A%20%20somme%3A%0A%20%20%20%20-%20prix%20.%20carottes%20*%201.5%20kg%0A%20%20%20%20-%20prix%20.%20champignons%20*%20500g%0A%20%20%20%20-%20prix%20.%20avocat%20*%203%20avocat%0A)

**Comment by simogeo** [comment]
ok, merci du retour même si je ne sais pas trop comment l'activer (?)

Au titre d'amélioration, il serait intéressant de pouvoir spécifier dans le modèle de données la possibilité de débugger, du style : 

```
dépenses primeur:
  debug:true
  somme:
    - prix . carottes * 1.5 kg
    - prix . champignons * 500g
    - prix . avocat * 3 avocat
```

Dans le but d'afficher dans la console, un truc du style : "dépenses primeur :  11,50 €"

Bonne journée,

**Comment by wiinxt** [comment]
Ca ne s'active pas, c'est un composant à intégrer dans une application React.
Pour le `debug` personnellement je trouve que c'est une bonne idée.
